src/oca_pre_commit_hooks/checks_odoo_module_po.py

```diff
@@ -39,51 +40,73 @@ class FormatStringParseError(StringParseError):
-        self.po_data = {
-            "filename": po_filename,
-            "filename_short": os.path.relpath(po_filename, top_path),
-            "data_section": os.path.basename(os.path.dirname(po_filename)),  # i18n or i18n_extra
-            "top_path": top_path,
-        }
+class ChecksOdooModulePO(BaseChecker):
+    def __init__(self, po_filename, enable, disable, autofix):
+        super().__init__(enable, disable, autofix=autofix)
+
+        self.top_path = utils.top_path(os.path.dirname(po_filename))
+        self.filename = utils.full_norm_path(po_filename)
+        self.filename_short = os.path.relpath(self.filename, self.top_path)
+        self.data_section = os.path.basename(os.path.dirname(self.filename))
+
+        self.po_data = []
+        self.original_contents = None
+        self.pretty_contents = None
+        self.file_error = None
+        self.needs_autofix = False
+
         try:
             with open(po_filename, encoding="UTF-8") as filename_obj:
-                # Do not use polib.pofile(self.po_data["filename"])
+                # Do not use polib.pofile()
                 # because raise the following error for PO files with syntax error:
                 # pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
                 # Traceback (most recent call last):
                 #     File "../polib.py", line 1474, in add
                 #     action = getattr(self, 'handle_%s' % next_state)
                 # ResourceWarning: unclosed file <_io.FileIO name='..' mode='rb' closefd=True>
-                polib_entries = polib.pofile(filename_obj.read())
-            self.po_data.update(
-                {
-                    "po": polib_entries,
-                    "file_error": None,
-                }
-            )
+                self.original_contents = filename_obj.read()
+
+            self.po_data = polib.pofile(self.original_contents)
         except (OSError, UnicodeDecodeError) as po_err:
-            self.po_data.update(
-                {
-                    "po": [],
-                    "file_error": po_err,
-                }
-            )
+            self.file_error = po_err
+
+    def _compute_pretty_contents(self):
+        self.po_data.sort(key=lambda entry: entry.msgid)
+        for entry in self.po_data:
+            if entry.msgid == entry.msgstr:
+                entry.msgstr = ""
+
+        self.pretty_contents = str(self.po_data)
+
+    def perform_autofix(self):
+        print(f"Fixing {self.filename_short} ⚒")
+        with open(self.filename, "w", encoding="utf-8") as po_fd:
+            po_fd.write(self.pretty_contents)
+
+    @utils.only_required_for_checks("po-pretty-format")
+    def check_po_pretty_format(self):
+        """* Check po-pretty-format
+        Check the following:
+        1. Entries sorted alphabetically
+        2. Lines are wrapped at 78 columns (same as Odoo)
+        3. Remove msgstr when it is the same as msgid
```

Comment:
I'd say "clear" or "leave empty" instead of "remove", because it's not actually removed, it's present but empty